### ISSUE TITLE
parsl.providers.base.JobStatus repr is broken (in python 3.11?)

### ISSUE THREAD

**Issue by benclifford** [Parsl/parsl#2652]
**Describe the bug**

**To Reproduce**
```
$ python3
Python 3.11.2 (tags/v3.11.2:878ead1, Mar  8 2023, 15:07:50) [GCC 12.2.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import parsl.providers.base as p
>>> str(p.JobState.COMPLETED)
'JobState.COMPLETED'
>>> repr(p.JobState.COMPLETED)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/local/lib/python3.11/enum.py", line 1191, in __repr__
    return "<%s.%s: %s>" % (self.__class__.__name__, self._name_, v_repr(self._value_))
                                                                  ^^^^^^^^^^^^^^^^^^^^
TypeError: descriptor '__repr__' requires a 'bytes' object but received a 'int'
>>> 
```
**Expected behavior**
repr should produce a sensible output - see behaviour of:

```
>>> import parsl.dataflow.states as s
>>> repr(s.States.running)
'<States.running: 2>'
```

**Environment**
python 3.11.2 source build
parsl master 8e4be00d6a148d476eac0b0973c5a93e6f27b63e
my laptop

**Comment by benclifford** [comment]
in python 3.10.6 from my underlying OS install, repr looks like this:

```
Python 3.10.6 (main, Mar 10 2023, 10:55:28) [GCC 11.3.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import parsl.providers.base as p
>>> str(p.JobState.COMPLETED)
'JobState.COMPLETED'
>>> repr(p.JobState.COMPLETED)
'<JobState.COMPLETED: 4>'
```

**Comment by benclifford** [comment]
crossref #2461

**Comment by benclifford** [comment]
This should be fixed by #2653
